airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeDestination.java

```diff
@@ -21,6 +29,27 @@ enum DestinationType {
   public SnowflakeDestination(final String airbyteEnvironment) {
     super(DestinationType.class, SnowflakeDestinationResolver::getTypeFromConfig,
         SnowflakeDestinationResolver.getTypeToDestination(airbyteEnvironment));
+    this.airbyteEnvironment = airbyteEnvironment;
+  }
+
+  @Override
+  public SerializedAirbyteMessageConsumer getSerializedMessageConsumer(final JsonNode config,
+                                                                       final ConfiguredAirbyteCatalog catalog,
+                                                                       final Consumer<AirbyteMessage> outputRecordCollector)
+      throws Exception {
+    log.info("destination class: {}", getClass());
+    // detect if running on internal staging for snowflake, if so run consumer2.
+    final boolean useAsyncSnowflake = false;
+    // final boolean useAsyncSnowflake = config.has("loading_method")
+    // && config.get("loading_method").has("method")
+    // && config.get("loading_method").get("method").asText().equals("Internal Staging");
```

Comment:
uncommenting these lines of code will all we will need to do to "turn on" the async version of the destination. as we do a feature flagged roll out, this will be the only diff in that feature branch.